SportHub.Business/Implementations/JwtService.cs

```diff
@@ -0,0 +1,132 @@
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+using Dapper;
+using Microsoft.Extensions.Configuration;
+using Microsoft.IdentityModel.Tokens;
+using MySql.Data.MySqlClient;
+using SportHub.Data;
+using SportHub.Data.DTOs;
+
+namespace SportHub.Business.Implementations;
+
+public class JwtService : IJwtService
+{
+    private readonly string _key;
+    private readonly string _connectionString;
+
+    public JwtService(IConfiguration config)
+    {
+        _key = config.GetSection("JwtSettings")["SecretKey"];
+        _connectionString = config.GetConnectionString("DefaultConnection");
+
+    }
+
+    public async Task<JwtResponse> GenerateTokens(User user)
+    {
+        var tokenHandler = new JwtSecurityTokenHandler();
+        var key = Encoding.ASCII.GetBytes(_key);
+        var tokenDescriptor = new SecurityTokenDescriptor
+        {
+            Subject = new ClaimsIdentity(new[]
+            {
+                new Claim(ClaimTypes.Email, user.Email),
+                new Claim(ClaimTypes.Name, user.FirstName + " " + user.SecondName),
+                new Claim(ClaimTypes.Role, user.IsAdmin ? "admin" : "user")
+            }),
+            Expires = DateTime.UtcNow.AddMinutes(1),
+            SigningCredentials = new SigningCredentials(new SymmetricSecurityKey(key),
+                SecurityAlgorithms.HmacSha256Signature)
+        };
+        
+        var accessToken = tokenHandler.CreateToken(tokenDescriptor);
+        
+        tokenDescriptor.Expires = DateTime.UtcNow.AddMinutes(2);
+        var refreshToken  = tokenHandler.CreateToken(tokenDescriptor);
+        _WriteTokenInDb(tokenHandler.WriteToken(refreshToken), user.Email);
```

Comment:
This method will become asyn you will need to await, also do note execute another method as parameter for other